Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs

```diff
@@ -402,80 +388,91 @@ private void EnsureAllTasksAreFinishedInChannel()
         }
 
-        private void MakeIngestionCallsFail()
+        private void SetupIngestionCallFail(Exception exception = null)
         {
-            _mockIngestion.CallShouldSucceed = false;
+            _mockIngestion
+                .Setup(ingestion => ingestion.Call(
+                    It.IsAny<string>(),
+                    It.IsAny<Guid>(),
+                    It.IsAny<IList<Log>>()))
+                .Returns((string appSecret, Guid installId, IList<Log> logs) =>
+                {
+                    var call = new ServiceCall(appSecret, installId, logs);
+                    call.SetException(exception);
+                    return call;
+                });
         }
 
-        private void MakeIngestionCallsSucceed()
+        private void SetupIngestionCallSucceed(string result = null)
         {
-            _mockIngestion.CallShouldSucceed = true;
+            _mockIngestion
+                .Setup(ingestion => ingestion.Call(
+                    It.IsAny<string>(),
+                    It.IsAny<Guid>(),
+                    It.IsAny<IList<Log>>()))
+                .Returns((string appSecret, Guid installId, IList<Log> logs) =>
+                {
+                    var call = new ServiceCall(appSecret, installId, logs);
+                    call.SetResult(result);
+                    return call;
+                });
         }
 
         private void SetupEventCallbacks()
         {
-            foreach (var sem in _eventSemaphores)
+            _eventSemaphores = new List<SemaphoreSlim>
             {
-                if (sem.CurrentCount != 0)
-                {
-                    sem.Release(sem.CurrentCount);
-                }
-            }
+                new SemaphoreSlim(0), new SemaphoreSlim(0), new SemaphoreSlim(0), new SemaphoreSlim(0), new SemaphoreSlim(0)
+            };
             _channel.SendingLog += (sender, args) => { _eventSemaphores[SendingLogSemaphoreIdx].Release(); };
             _channel.SentLog += (sender, args) => { _eventSemaphores[SentLogSemaphoreIdx].Release(); };
             _channel.FailedToSendLog += (sender, args) => { _eventSemaphores[FailedToSendLogSemaphoreIdx].Release(); };
             _channel.EnqueuingLog += (sender, args) => { _eventSemaphores[EnqueuingLogSemaphoreIdx].Release(); };
             _channel.FilteringLog += (sender, args) => { _eventSemaphores[FilteringLogSemaphoreIdx].Release(); };
         }
 
-        private bool FailedToSendLogOccurred(int numTimes, int waitTime = DefaultWaitTime)
-        {
-            return EventWithSemaphoreOccurred(_eventSemaphores[FailedToSendLogSemaphoreIdx], numTimes, waitTime);
-        }
+        private void VerifySendingLog(int expectedTimes, int waitTime = DefaultWaitTime) =>
+            Assert.AreEqual(expectedTimes, EventWithSemaphoreOccurred(_eventSemaphores[SendingLogSemaphoreIdx], expectedTimes, waitTime),
+                $"Failed on verify {nameof(Channel.SendingLog)} event call times");
```

Comment:
It was very uncomfortable to look for what was wrong without these messages. Cosmetic - yes, but.. since I'm changing this file and I need it for the convenience of debugging.. 
I think that in the end, it will save time not only for me